packages/workflow-core/src/lib/utils/definition-validator/definition-validator.ts

```diff
@@ -0,0 +1,17 @@
+import { StateMachine } from 'xstate';
+
+
+import {WorkflowExtensions} from "../../types";
+import {statesValidator, validateTransitionOnEvent} from "./states-validator";
+import {extensionsValidator} from "./extensions-validator";
+export const definitionValidator = (
+  definition: {
+    states: StateMachine<any, any, any>['states'],
+    initial: string
+  },
+  extensions?: WorkflowExtensions,
+) => {
+  validateTransitionOnEvent(Object.keys(definition.states), 'NULL_AS_UNINITIATED_STATE', definition.initial);
+  statesValidator(definition.states)
+  extensions && extensionsValidator(extensions, definition.states)
```

Comment:
Consider using `if` for invoking functions and conditional logic and `||` or `&&` etc. for returns and boolean variables. Expression like this tend to change overtime and become unreadable due to formatting.
```suggestion
  if (extensions) {
       extensionsValidator(extensions, definition.states)
   }
```

```suggestion
  if (!extensions) return;
  
  extensionsValidator(extensions, definition.states)
```